### PR TITLE
Add server context helper

### PR DESCRIPTION
**Pull request by acroz**
This provides a convenience interface for getting information about the server context that will be used by the resource-based interface in a subsequent PR.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -33,6 +33,7 @@
         "enum34; python_version<'3.4'",
         # Install marshmallow with 'reco' (recommended) extras to ensure a
         # compatible version of python-dateutil is available
+        "attr",
```
Comment (zblz) [inline setup.py]: `attr` is, confusingly, another package:
```suggestion
        "attrs",
```

File: faculty/context.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+def _get_environ_as_type(key, cls):
+    try:
+        return cls(os.environ[key])
+    except KeyError:
+        # Not in environment
+        return None
+    except ValueError:
+        # Badly formatted
+        template = (
+            "Error interpreting badly formatted environment varaible {}={}"
```
Comment (srstevenson) [inline faculty/context.py]:
```suggestion
            "Error interpreting badly formatted environment variable {}={}"
```

Comment (acroz) [reply faculty/context.py]: thanks 👍 - good spot

File: setup.py (hunk 1 above)
Comment (acroz) [reply setup.py]: Well spotted - this only worked in tests as `pytest` depends on `attrs`.

**Pull request merged**